sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -262,13 +354,24 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
 
     val shuffleStages = collectShuffleStages(plan)
 
-    if (shuffleStages.length == 2) {
+    if (shuffleStages.length >= 1) {
```

Comment:
Does this mean we will also handle multi-table join (e.g. three-table join) ?

Reply:
> Does this mean we will also handle multi-table join (e.g. three-table join) ?

Since broadcast nested loop join only can have one side shuffle exchange, but sort merge join with two